### PR TITLE
Backport minor changes from `feat-fast-sync`.

### LINKED ISSUE #2274: Backport minor changes from feat-fast-sync to dev
Description:

* [ ] Disable `bind_to_real_network_interface` test for MacOS
* [ ] Remove `HandlingEraSupervisor`
* [ ] Fix `EraEnd`/`EraReport` confusion.
* [ ] Extract `Participating::handle_get_request`
* [ ] ~`Deploy::equals_ignoring_is_valid` for testing~ (might change; maybe we should just call `is_valid`, like on `dev`)
* [ ] ~`BlockWithMetadata`, `BlockHeaderWithMetadata`~ (too entangled with other fetcher and storage changes)
* [ ] ~`test_joiner_network` changes (ignored on dev anyway)~ (specific to the "immediate switch block" change)

Reporter: @afck 
Approver: @Fraser999

### PR DESCRIPTION
**Pull request by afck**
Backport minor changes from the fast sync feature branch:

* Remove the `EraSupervisorHandlingWrapper`; that pattern was only causing confusion.
* Rename `active_eras` to `open_eras`: These eras are kept in memory, and some only partly, but most of them are not active in the sense that new units are being created.
* Disable the `bind_to_real_network_interface` test on MacOS, where it doesn't work.
* Clear up `EraEnd`/`EraReport` confusion: We had mixed up the terminology in a few places.
* Extract `Participating::handle_get_request`. We'll probably want to extract more methods like this in the future, but here we go only as far as on the fast sync branch.

Closes #2274.

### CODE REVIEW

**Review by xcthulhu — APPROVED** [review]

**Review by goral09 — APPROVED** [review]

File: node/src/types/block.rs [hunk 1]
```diff
@@ -576,28 +576,31 @@ impl FromBytes for BlockHash {
...
+    /// The validators for the upcoming era and their respective weights.
     next_era_validator_weights: BTreeMap<PublicKey, U512>,
 }
 
 impl EraEnd {
-    pub fn new(
-        era_report: EraReport,
-        next_era_validator_weights: BTreeMap<PublicKey, U512>,
-    ) -> Self {
+    fn new(era_report: EraReport, next_era_validator_weights: BTreeMap<PublicKey, U512>) -> Self {
```
Comment (xcthulhu) [inline node/src/types/block.rs:586]: Why change the formatting here?

Comment (afck) [reply node/src/types/block.rs:586]: I removed the `pub`, and rustfmt did the rest.

**Comment by afck** [comment]
bors merge

**Comment by afck** [comment]
bors merge

**Pull request merged**